### DISCUSSION TITLE
Using access_token in Custom API Hook

### DISCUSSION THREAD

**Discussion by iherwig** [directus/directus#12304] [Archive]
Is it possible to get the access_token of the current user inside a custom api hook? This would be usefull to call a custom api endpoint which requires an authenticated user.

**Comment by rijkvanzanten** [comment]
You wanna call a Directus API (custom) endpoint from a custom hook? I would recommend just using functions / services directly instead of doing another network call from the hook itself 🤔

**Reply by iherwig** [reply]
@rijkvanzanten Thanks for the suggestion. I implemented it by using the `emitter` instance that is passed to the register functions of the hook and endpoint. The hook is sending an event that the endpoint receives.

**Comment by iherwig** [comment]
I have the endpoint already and tried to call it in a regular directus webhook. Since this didn't work (because of missing authentication) I thought about implementing a custom hook that could handle this. Of course I could also implement the endpoint functionality in the custom hook, but i wanted to keep things decoupled.

**Reply by rijkvanzanten** [reply]
Ahh gotcha, that makes sense 👍🏻
